### PR TITLE
intergration spec

### PR DESCRIPTION
**Pull request by monicah79**
In this repo: Add integration specs for views and fix n+1 problems

In this branch, we implemented the following:

Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.
before
image
after
image

-Used Capybara to write integration tests for each view in your project.

*User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

**Comment by monicah79** [comment]
Hey am leaving this comment because I had done this project and it had requested changes bt it gave me too much trouble so I decided to move forward to other projects when I was merging the devise branch I ended up merging the branch that wasnt approved bt I have done everything the test and the n+1 project as per the project requirement I really dint have many options. I was to create a pull request on the same branch but I realized if I work on that and I have other branches with the current work it would affect the project performance so I figured out it was wise for me to create a new branch with the current project kindly consider my project and review.
Thanks

**Comment by monicah79** [comment]
I push because i realised it had a conflict

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -3,25 +3,34 @@
...
+    it 'can see the number of posts each user has written.' do
+      visit '/user'
+      page.has_content?(user.posts_counter)
+    end
+
+    it "When I click on a user, I am redirected to that user's show page." do
+      User.create(name: 'Jame', email: 'john.james+2@example.com', bio: "Hello, I'm james!", posts_counter: 3,
+                  password: 211_111)
+      visit '/user'
+    end
```
Comment (rloterh) [inline spec/views/user_index_spec.rb:35]: - [ ] Let's ensure to include test for the below scenarios for _user index_ views: 
   * I can see the number of posts each user has written
   * When I click on a user, I am redirected to that user's show page

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -43,19 +22,28 @@
...
+  it "I can see a button that lets me view all of a user's posts." do
+    visit 'user'
+    page.has_button?('See all posts')
+  end
+
+  it "When you go see all posts, it redirects me to the user's post's index page." do
+    visit 'user'
+    visit 'user/1post'
+    page.has_content?(user.name)
+  end
```
Comment (rloterh) [inline spec/views/user_show_spec.rb]: - [ ] Let's ensure to include test for the below scenarios for _user show_ views: 
   * I can see the user's profile picture
   * I can see the user's username
   * I can see the user's first 3 posts
   * When I click a user's post, it redirects me to that post's show page

File: spec/views/post_index_spec.rb [hunk 3]
```diff
@@ -23,24 +24,23 @@
...
-    expect(page).to have_content("Number of posts: #{user.posts_counter}")
+  it "I can see a post's title." do
+    visit '/user/1/post'
+    page.has_content?(post.title)
   end
 
   it 'has a link to view each post' do
     visit '/user/1/post'
+    page.has_content?(post.text)
   end
```
Comment (rloterh) [inline spec/views/post_index_spec.rb]: - [ ] Let's ensure to include test for the below scenarios for _post index_ views: 
   * I can see the user's profile picture
   * I can see the first comments on a post
   * I can see how many comments a post has
   * I can see how many likes a post has
   * I can see a section for pagination if there are more posts than fit on the view
   * When I click on a post, it redirects me to that post's show page

File: spec/views/post_show_spec.rb [hunk 4]
```diff
@@ -49,8 +40,9 @@
...
   end
 
-  it 'displays post author name' do
-    expect(page).to have_content(user.name)
+  it "I can see the user's name." do
+    visit '/user/1/post'
+    page.has_content?(user.name)
   end
 
   it 'counts total number of comments' do
```
Comment (rloterh) [inline spec/views/post_show_spec.rb:48]: - [ ] Let's ensure to include test for the below scenarios for _post show_ views: 
   * I can see the post's title
   * I can see who wrote the post
   * I can see how many likes it has
   * I can see the post body
   * I can see the username of each commentor
   * I can see the comment each commentor left

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Monicah, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦  ⛪ 🚜   🏃‍♀️  🌴  🏫 🌳 🏘️ 

## Highlights 👍  
- N + 1 problem addressed!
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉
### Highlights
- Correct workflow was used ✔️
- Good project structure ✔️
- Well-organized and clean project ✔
thanks-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**